packages/core/src/routes/session.test.ts

```diff
@@ -274,6 +431,270 @@ describe('sessionRoutes', () => {
     });
   });
 
+  describe('POST /session/register/username-password', () => {
+    it('assign result and redirect', async () => {
+      const response = await sessionRequest
+        .post('/session/register/username-password')
+        .send({ username: 'username', password: 'password' });
+      expect(insertUser).toHaveBeenCalledWith(
+        expect.objectContaining({
+          id: 'user1',
+          username: 'username',
+          passwordEncrypted: 'user1_password_user1',
+          passwordEncryptionMethod: 'SaltAndPepper',
+          passwordEncryptionSalt: 'user1',
+        })
+      );
+      expect(response.body).toHaveProperty('redirectTo');
+      expect(getProvider().interactionResult).toHaveBeenCalledWith(
+        expect.anything(),
+        expect.anything(),
+        expect.objectContaining({ login: { accountId: 'user1' } }),
+        expect.anything()
+      );
+    });
+
+    it('throw error if username not valid', async () => {
+      const response = await sessionRequest
+        .post('/session/register/username-password')
+        .send({ username: '_', password: 'password' });
+      expect(response.statusCode).toEqual(400);
+    });
+
+    it('throw error if username exists', async () => {
+      const response = await sessionRequest
+        .post('/session/register/username-password')
+        .send({ username: 'username1', password: 'password' });
+      expect(response.statusCode).toEqual(422);
```

Comment:
should we align the error code of 422? as current entity not found error throws 400 for the rest routes